src/System.Device.Gpio/System/Device/Gpio/GpioDriver.cs

```diff
@@ -43,6 +43,23 @@ public abstract class GpioDriver : IDisposable
         /// <param name="mode">The mode to be set.</param>
         protected internal abstract void SetPinMode(int pinNumber, PinMode mode);
 
+        /// <summary>
+        /// Sets the mode to a pin and sets an initial value for an output pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        /// <param name="initialValue">The initial value if the <paramref name="mode"/> is output. The driver will do it's best to prevent glitches to the other value when
+        /// changing from input to output.</param>
+        protected internal virtual void SetPinMode(int pinNumber, PinMode mode, PinValue initialValue)
+        {
+            // TODO: Find out which driver support this atomically
```

Comment:
Haven't you already figured this out on this PR? If so we can remove the comment.